integ-test/src/test/java/org/opensearch/sql/ppl/WildcardQueryIT.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ppl;
+
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_BEER;
+
+import java.io.IOException;
+import org.json.JSONObject;
+import org.junit.Test;
+
+public class WildcardQueryIT extends PPLIntegTestCase {
+
+  @Override
+  public void init() throws IOException {
+    loadIndex(Index.BEER);
+  }
+
+  @Test
+  public void wildcard_query_function_test() throws IOException {
+    String query = "source=" + TEST_INDEX_BEER + " | where wildcard_query(Tags, 't*')";
+    JSONObject result = executeQuery(query);
+    assertEquals(10, result.getInt("total"));
+  }
+
+  @Test
+  public void test_wildcard_query_sql_wildcard_conversion() throws IOException {
```

Comment:
Would be nice to see some more data validated other than totals. You can use `verifyDataRows` for returned rows. You can also use the `HEAD` command to limit the returned rows.

Reply:
In addition, can you clarify which are valid wildcard characters?

Between these tests and the user documentation, I'm unclear what they are. Documentation refers to OpenSearch documentation which only lists `*` and `?` yet the comments in the tests below imply that `%` and `_` are also supported.

Reply:
Fixed in a2ca906329ab194b845b8078eafeae48eb4bf1c9. The legacy engine supports * and ? as well as sql wildcards % and _ as can be seen in https://github.com/opensearch-project/sql/blob/7bb52a8fd1fa00fe16f5a71c14c2a8e836c9b8e1/opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/WildcardQuery.java#L26. I added more details in test comments and the docs.